pandas/tests/dtypes/test_inference.py

```diff
@@ -1455,11 +1455,18 @@ def test_is_string_array(self):
         assert lib.is_string_array(
             np.array(["foo", "bar", pd.NA], dtype=object), skipna=True
         )
-        # NaN is not valid for string array, just NA
-        assert not lib.is_string_array(
+        assert lib.is_string_array(
```

Comment:
IIUC from https://github.com/pandas-dev/pandas/pull/41412#discussion_r637433431, we now call ensure_string_array in the constructor whereas previously is_string_array was called. so why is lib.is_string_array changing.

Reply:
lib.is_string_array is used for inferring dtypes elsewhere I think, and we want ndarrays with strings and np.nan/None to be inferred as string in addition to ndarrays with only strings/pd.NA.

Reply:
right. so is `is_string_array` called before passing the values to the `StringArray` constructor? The reason I ask is that a reasonably contained behavior change to StringArray could be backported to 1.3.x during the rc period

will hopefully get a chance to look in more detail soon, but if you could summarize why it's changing that'll be great.

Reply:
No, it isn't called anymore in this PR. However, ``lib.is_string_array`` is called in ``lib.infer_dtype`` (https://github.com/pandas-dev/pandas/blob/09296b84ef753828f251029e92b8cabb1c277437/pandas/_libs/lib.pyx#L1300). Because we now accept other NA values in StringArray, ``is_string_array`` should also return true when those values are encountered in a string array for consistency reasons.